components/fxa-client/src/oauth.rs

```diff
@@ -314,7 +313,40 @@ impl FirefoxAccount {
         // When our keys change, we might need to re-register device capabilities with the server.
         // Ensure that this happens on the next call to ensure_capabilities.
         self.state.device_capabilities.clear();
+        Ok(())
+    }
 
+    /// Typically called during a password change flow.
+    /// Invalidates all tokens and fetches a new refresh token.
+    /// Because the old refresh token is not valid anymore, we can't do like `handle_oauth_response`
+    /// and re-create the device, so it is the responsibility of the caller to do so after we're
+    /// done.
```

Comment:
Is there a potential future refactor that *would* allow us to handle this automatically, such as if we kept a local copy of the device record?

Reply:
Absolutely! We could have just like profile a `last_seen_profile` for non critical uses like this.